v2/pkg/engine/datasource/graphql_datasource/graphql_subscription_client.go

```diff
@@ -30,12 +30,25 @@ type subscriptionClient struct {
 	hashPool                   sync.Pool
 	handlers                   map[uint64]ConnectionHandler
 	handlersMu                 sync.Mutex
-	wsSubProtocol              string
 	onWsConnectionInitCallback *OnWsConnectionInitCallback
 
 	readTimeout time.Duration
 }
 
+type InvalidWsSubprotocolError struct {
+	Message string
```

Comment:
Instead of putting a generic string here, create public fields for expected and actual value, then this can be parsed in the Router and we don't have to deal with a generic string.

Reply:
Note: we will not parse it in the router

we just will log it

When websocket connection is established it lives in a separate goroutine, and we won't be able to communicate this error to the router itself

Reply:
No, this error can be returned. This is while creating the wsConnectionHandler. The thing happening inside the go routine is StartBlocking function stuff right?